### PR TITLE
Pr-SpearLanceEnemy(I renamed the branch but for some reason it's not changed)

### PR DESCRIPTION
**Pull request by abbclas**

### CODE REVIEW

**Comment by abbclas** [comment]
the number of starting boids is now editable in the inspector if you go to the player scene, the rat enemy isn't done , so for now I'm just going to merge the spear guys.

File: app/Scripts/EnemySpawner.gd [hunk 1]
```diff
@@ -1,11 +1,11 @@
 extends Node2D
 
-onready var _player = get_node("/root/Main/Player")
+onready var _player = get_node("../YSort/Player")
```
Comment (noh0st) [inline app/Scripts/EnemySpawner.gd:3]: what is ysort?

File: app/Scenes/Main.tscn [hunk 2]
```diff
@@ -32,11 +26,33 @@ parameters/playback = SubResource( 1 )
...
+}
+
+[node name="AnimationPlayer" parent="YSort/Player" index="2"]
+root_node = NodePath("../../..")
+
+[node name="AnimationTree" parent="YSort/Player" index="3"]
+parameters/playback = SubResource( 1 )
+
+[node name="SpearLanceEnemy" parent="." instance=ExtResource( 10 )]
+position = Vector2( 210, -129 )
```
Comment (noh0st) [inline app/Scenes/Main.tscn:50]: can we use the enemy manager to spawn this instead?

File: app/Scripts/RockTurtle.gd [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		
+		attackTimer.start(0)
+		print(TargetDir, area)
+		
+
+
+func _on_TurtleVision_area_exited(area):
+	if area == Target:
+		print("exit")
+		if HasTarget == true:
```
Comment (noh0st) [inline app/Scripts/RockTurtle.gd:29]: we don't need a separate HasTarget variable, you can just check if `is_instance_valid(Target)` I believe

File: app/Scripts/SpearLanceEnemy.gd [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+	_random_direction()
+	set_meta("Enemy", false)
+	isAttackingAnimation = false
+	Velocity = TargetPos - self.position
+	_animationPlayer.play("Idle")
+	Move()
+	PlayerStats.connect("Death", self, "TargetDead")
+func _on_Vision_area_entered(area): #if you dont have target, set a target
+	$Vision/CollisionShape2D.disabled = true
+	if HasTarget == false:
```
Comment (noh0st) [inline app/Scripts/SpearLanceEnemy.gd:26]: same here, HasTarget is redundant i believe

File: app/Scripts/SpearLanceEnemy.gd [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+##########
+
+func Attack():
+	#attacking - stops the attacks, the hitboxes are controlled through the animation player
+	if Target != null:
+		Velocity=Vector2.ZERO
+		Dir = Velocity.normalized()
+		AttackDir()
+		_attackTimer.start(0)
+	if Target == null:
```
Comment (noh0st) [inline app/Scripts/SpearLanceEnemy.gd:85]: nit: else:

File: app/Scripts/SpearLanceEnemy.gd [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+		if HasTarget == true :
+			if MoveTowardsTarget:
+				if(!isAttackingAnimation):
+					Velocity = Target.get_parent().position - self.position 
+					MovementDir()
+					move_and_slide(Dir * 100)
+			Dir = Velocity.normalized()
+			
+			
+	if HasTarget == false:
```
Comment (noh0st) [inline app/Scripts/SpearLanceEnemy.gd:100]: this should be an else

File: app/Scripts/SpearLanceEnemy.gd [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+			Target = area
+			
+			MoveTowardsTarget = true
+			HasTarget = true
+			_patrolTimer.stop()
+			
+		
+func _physics_process(delta): #Target death
+	if Target != null && Target.get_parent().has_meta("Boid"):
+		if !BoidsGlobal.AllBoidsArray.has(Target.get_parent()):
```
Comment (noh0st) [inline app/Scripts/SpearLanceEnemy.gd:38]: why do we need a global boids array? Isn't it enough to know whether the Target object is valid or not?

File: app/Scripts/EnemySpawner.gd [hunk 8]
```diff
@@ -1,11 +1,11 @@
...
 onready var _enemy_manager = get_node("/root/Main/EnemyManager")
 onready var _timer: Timer = $Timer
 
 const SPAWN_RADIUS = 250
-const MAX_COUNT = 5
+const MAX_COUNT = 0
```
Comment (noh0st) [inline app/Scripts/EnemySpawner.gd:8]: we are we setting this to 0?

File: app/Scripts/enemy.gd [hunk 9]
```diff
@@ -40,14 +51,15 @@ func _process_walking(delta: float):
...
+			#choose direction and switch to walking
 			_walk_direction = _random_normalized_direction()
 			_current_state = State.WALKING
 			_timer.start(rand_range(2, 5))
+			pass
 		State.WALKING:
 			# stop walking
 			_current_state = State.IDLE
-			_timer.start(rand_range(2, 5))
+			#_timer.start(rand_range(2, 5))
```
Comment (noh0st) [inline app/Scripts/enemy.gd:62]: if we aren't spawning enemies then i think we should revert these changes. Fine to keep all the new images, but all the other changes that are unverifiable should ideally not be included in the PR. (otherwise main will be full of confusing unfinished code)

File: app/Scripts/EnemySpawner.gd (hunk 8 above)
Comment (abbclas) [reply app/Scripts/EnemySpawner.gd:8]: just so rats dont spawn

File: app/Scenes/Main.tscn (hunk 2 above)
Comment (abbclas) [reply app/Scenes/Main.tscn:50]: i think so

File: app/Scripts/SpearLanceEnemy.gd (hunk 7 above)
Comment (abbclas) [reply app/Scripts/SpearLanceEnemy.gd:38]: for some reason, it doesn't understand when when queue_free() the boids so I have to do that to tell it the boid has died
